### PR TITLE
feat: initial vector implementation

### PR DESCRIPTION
**Pull request by Ziinc**
Adding in vector, for handling postgrest, postgres, and gotrue logging pipeline.

### CODE REVIEW

File: internal/utils/templates/vector.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    logflare_auth:
+        type: "http"
+        inputs:
+            - auth_logs
+        encoding:
+            codec: "json"
+        method: "post"
+        request:
+            retry_max_duration_secs: 10
+        uri: "http://logs:4000/api/logs?source_name=gotrue.logs.prod&api_key=gd9tDpgiCVJe"
```
Comment (sweatybridge) [inline internal/utils/templates/vector.yaml:58]: substitute with template variables here too

File: internal/start/templates/vector_config [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      - postgrest_journald
+    condition: >-
+      !starts_with!(.message, "+") && !starts_with!(.message, "INFO:") && !contains!(.message, "Admin server listening")
+
+  gotrue_to_object:
+    inputs:
+      - gotrue_journald
+    type: remap
+    source: |2-
+                .project = "project_ref"
```
Comment (sweatybridge) [inline internal/start/templates/vector_config:26]: Does this need to replaced by template variable?

File: internal/start/templates/vector_config [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+sinks:
+  http_gotrue:
+    type: http
+    inputs:
+      - gotrue_to_object
+    encoding:
+      codec: json
+    compression: none
+    uri: >-
+      https://api.logflare.app/logs?api_key=logflare_api_key&source=logflare_source
```
Comment (sweatybridge) [inline internal/start/templates/vector_config:101]: Do we need to point this to local logflare?

File: internal/utils/templates/vector.yaml [hunk 4]
```diff
@@ -0,0 +1,109 @@
+api:
+    enabled: true
+    address: "0.0.0.0:9001"
+
+sources:
+    docker_syslog:
+        type: "syslog"
+        address: "0.0.0.0:9000"
```
Comment (sweatybridge) [inline internal/utils/templates/vector.yaml:8]: should this be port 8000 to match docker syslog port?

File: internal/start/templates/vector_config (hunk 2 above)
Comment (Ziinc) [reply internal/start/templates/vector_config:26]: nope, for local dev, project ref defaults to `default`, so not necessary

**Review by sweatybridge — APPROVED** [review]

**Comment by sweatybridge** [comment]
Vector ingestion verified to be working from logflare ui. 

Screenshot 2023-03-31 at 10 15 15 AM

Studio ui seems to have trouble querying which will resolve later

Screenshot 2023-03-31 at 9 55 51 AM

**Pull request merged**